### ISSUE TITLE
Stopping in progress jobs

### ISSUE THREAD

**Issue by herecydev** [leosperry/Chroniton#12]
How does this library cater for jobs that are in progress when a Stop is initiated. It would be great to have a mechanism like `Task StopAndWait()` that stops all scheduled jobs allows all in progress jobs to finish. Is this possible here?

**Comment by leosperry** [comment]
If you look at the Stop() method in Singularity.cs, you'll see that it essentially does that. First it sets a boolean flag to prevent more jobs from getting started. Then, it waits for all currently executing jobs to finish. This logic will change slightly in the next version, but there will always be an opportunity for currently executing jobs to finish. You can call the Stop() method yourself or optionally allow the deconstructor to call it.

**Comment by herecydev** [comment]
I believe it [waits on the scheduling thread](https://github.com/leosperry/Chroniton/blob/master/ChronitonShared/Singularity.cs#L135). But not the tasks in progress. Oddly you're using Task.WaitAll() on a single task, did you mean that to be `Task.WaitAll(_tasks);`?

A quick example:

``` c#
var singularity = Singularity.Instance;
            var schedule = new EveryXTimeSchedule(TimeSpan.FromSeconds(6));
            var job = new SimpleParameterizedJob<long>(async (id, dt) =>
            {
                Console.WriteLine("Sleeping");
                await Task.Delay(5000);
                Console.WriteLine(id);
                Console.WriteLine(id);
                Console.WriteLine(id);
            });
            var sch = singularity.ScheduleParameterizedJob(schedule, job, 123, true);

            singularity.Start();

            Thread.Sleep(1000);

            singularity.Stop();

            Console.WriteLine("Stopped");

            Console.ReadLine();
```

Gives me the following:

![image](https://cloud.githubusercontent.com/assets/11328618/19283092/92cad832-8fe8-11e6-934f-dc90961f90b5.png)

I'm fairly confident that might be a typo, based on your use of Task.WaitAll(). but I believe the other stuff (cancelling scheduled etc) is correct, as there is no more further input.

**Comment by leosperry** [comment]
You are correct.  I looked at this issue in another branch, but it has not been merged yet.

**Comment by herecydev** [comment]
Ah, now I see what you're referring to. Can I help in expediting that, or is it stable enough to release an RC perhaps?

**Comment by leosperry** [comment]
Yes please. It'd be great to have more contributors.

**Comment by herecydev** [comment]
Sure, do you have a list of the remaining issues. Also it would be better if the folder structure was a little cleaner. David Fowler [has a great example](https://gist.github.com/davidfowl/ed7564297c61fe9ab814) of project layout. You only need 1 source project, a test project, and maybe a couple of examples, that you have. The other projects feel superfluous. I can put something together if you would like?

**Comment by herecydev** [comment]
A little taste can be [found here](https://github.com/leosperry/Chroniton/pull/13)
